### PR TITLE
Feature/29 write integration tests

### PR DESCRIPTION
**Pull request by mirkojotic**
Hi,

Thought to give it a try and write some tests. I don't really need this to be merged, I just wanted to check if I'm going in the right direction ( likely not :D ).

### CODE REVIEW

File: test/SimpleTest.php [hunk 1]
```diff
@@ -2,15 +2,128 @@
...
+    }
+
+    /**
+     * @expectedException Bschmitt\Amqp\Exception\Configuration
+     */
+    public function testIfEmptyExchangeThrowsAnException()
+    {
+        $this->publisherMock->mergeProperties(['exchange' => '']);
+        $this->publisherMock->shouldReceive('connect');
+        // will throw an Exception
```
Comment (stevenklar) [inline test/SimpleTest.php]: I already expect that by the docblock. Good comment but read redundant for me.

File: test/SimpleTest.php [hunk 2]
```diff
@@ -2,15 +2,128 @@
...
+            'queue' => $queueName, 
+            'queue_force_declare' => true,
+            'routing' => $routing
+        ]);
+        $this->channelMock->shouldReceive('exchange_declare');
+        $this->publisherMock->shouldReceive('connect');
+
+        $this->channelMock->shouldReceive('queue_declare')
+                          ->with(
+                            $queueName,
```
Comment (stevenklar) [inline test/SimpleTest.php]: 2 spaces -> 4 spaces
:-)

File: test/SimpleTest.php [hunk 3]
```diff
@@ -2,15 +2,128 @@
...
+        $this->setProtectedProperty('\Bschmitt\Amqp\Publisher', $this->publisherMock, 'connection', $this->connectionMock);
+    }
+
+    protected function tearDown()
+    {
+        // necessary for Mockery to check if methods were called and with what arguments
+        Mockery::close();
+    }
 
     public function testClassInstance()
```
Comment (stevenklar) [inline test/SimpleTest.php]: Empty tests can be removed.

File: test/SimpleTest.php [hunk 4]
```diff
@@ -2,15 +2,128 @@
...
+        $this->connectionMock = Mockery::mock('\PhpAmqpLib\Connection\AMQPSSLConnection');
+        // channel and connection are both protected and without changing the source this was the only way to mock them
+        $this->setProtectedProperty('\Bschmitt\Amqp\Publisher', $this->publisherMock, 'channel', $this->channelMock);
+        $this->setProtectedProperty('\Bschmitt\Amqp\Publisher', $this->publisherMock, 'connection', $this->connectionMock);
+    }
+
+    protected function tearDown()
+    {
+        // necessary for Mockery to check if methods were called and with what arguments
+        Mockery::close();
```
Comment (stevenklar) [inline test/SimpleTest.php]: Good thing! :+1:

File: test/SimpleTest.php [hunk 5]
```diff
@@ -2,15 +2,128 @@
...
+use Bschmitt\Amqp\Publisher;
+use \Mockery;
+use Illuminate\Config\Repository;
+
 /**
  * @author Björn Schmitt <code@bjoern.io>
  */
 class SimpleTest extends \PHPUnit_Framework_TestCase
 {
+    private $REPOSITORY_KEY = 'amqp';
```
Comment (stevenklar) [inline test/SimpleTest.php]: What do you think about a constant instead? This shouldn't change for the tests.

**Comment by mirkojotic** [comment]
@stevenklar thank you for the feedback! Everything you said makes sense, I'll make necessary modifications and then continue with Message and Request classes.

**Pull request merged**